src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -0,0 +1,1443 @@
+// that should be copied over
+
+internal class MyBase
+{
+    int Method()
+    {
+        return 1 + 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestWithInterface()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : ITest
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : MyBase, ITest
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int Method()
+    {
+        return 1 + 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [ConditionalFact(AlwaysSkip = "https://github.com/dotnet/roslyn/issues/45977")]
+        public async Task TestRegion()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    #region MyRegion
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+
+    void OtherMethiod() { }
+    #endregion
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+
+    #region MyRegion
+
+    void OtherMethiod() { }
+    #endregion
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    #region MyRegion
+    int Method()
+    {
+        return 1 + 1;
+    }
+    #endregion
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(
+                input,
+                expected,
+                dialogSelection: new[] { ("Method", false) });
+        }
+
+        [Fact]
+        public async Task TestMakeAbstract_SingleMethod()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+    override int Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal abstract class MyBase
+{
+    private abstract global::System.Int32 Method();
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(
+                input,
+                expected,
+                dialogSelection: new[] { ("Method", true) });
```

Comment:
this doesn't read well.

Reply:
Added `MakeSelection` and `MakeAbstractSelection` methods to clean this up